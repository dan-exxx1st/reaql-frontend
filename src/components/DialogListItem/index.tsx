import React, { FC } from 'react';
import StyledDialogListItem from './style';

const DialogListItem: FC = () => {
	return <StyledDialogListItem>Dialog list item</StyledDialogListItem>;
};

export default DialogListItem;
