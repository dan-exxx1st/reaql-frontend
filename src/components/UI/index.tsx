export { default as Avatar } from './Avatar';
export { default as Button } from './Button';
export { default as Circle } from './Circle';
export { default as SkeletonLoader } from './SkeletonLoader';
export { default as TextField } from './TextField';
export { default as Typography } from './Typography';
export { default as Wrapper } from './Wrapper';
export { default as CheckMark } from './CheckMark';
export { default as Icon } from './Icon';
export { default as CheckBox } from './CheckBox';
export { default as SystemMessage } from './SystemMessage';
export { default as Spinner } from './Spinner';
