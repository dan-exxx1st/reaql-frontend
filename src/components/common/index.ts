export { default as DialogHeader } from './DialogHeader';
export { default as ContactItem } from './ContactsItem';
export { default as UserHeader } from './UserHeader';
export { default as DialogTextField } from './DialogTextField';
export { default as MessageItem } from './MessageItem';
export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';
export { default as MessageSearch } from './MessageSearch';
