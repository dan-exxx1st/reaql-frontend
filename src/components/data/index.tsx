export { default as MessageListWithData } from './MessageListWithData';
export { default as DialogHeaderWithData } from './DialogHeaderWithData';
