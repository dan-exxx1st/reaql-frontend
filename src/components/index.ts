export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
