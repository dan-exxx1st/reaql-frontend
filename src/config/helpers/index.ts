import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
