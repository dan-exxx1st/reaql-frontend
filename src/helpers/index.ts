import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
