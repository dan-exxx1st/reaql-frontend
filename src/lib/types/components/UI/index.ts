import { MouseEventHandler } from 'react';
import Theme from 'helpers/styled';
import { ColorType, IconName } from 'lib/types';
import { TypographyVariantType, TypographyComponentType } from 'lib/types/ui';

export interface IAvatarProps {
    src?: HTMLImageElement['src'];
    alt?: string;
    className?: string;
}

export interface IButtonProps {
    className?: string;
    theme: typeof Theme;
    onClick?: MouseEventHandler;
    variant: 'filled' | 'outlined';
    color: ColorType;
}

export interface ICheckMarkProps {
    isChecked: boolean;
    isDouble: boolean;
}

export interface ICircleProps {
    color: ColorType;
    iconName?: IconName;
    text?: string | number;
    width?: string;
    height?: string;
    cursor?: string;
    className?: string;
    onClick?: () => void;
}

export interface IIconProps {
    iconName: IconName;
    className?: string;
}

export interface ISkeletonLoaderProps {
    variant?: 'avatar' | 'block';
    width?: string;
    height?: string;
    className?: string;
}

export interface ITextFieldProps {
    onChange?: (value: string) => void;
    value?: string;
    className?: string;
    fieldSize: 'small' | 'medium' | 'large';
    color: ColorType;
    icon?: IconName;
    width?: string;
    height?: string;
    wrapperWidth?: string;
    placeholder?: string;
    type?: string;
}

export interface ITypographyProps {
    className?: string;
    variant: TypographyVariantType;
    component?: TypographyComponentType;
    color?: string;
    theme: typeof Theme;
}

export type { IWrapperProps } from './WrapperTypes';

export interface ICheckBoxProps {
    onClick?: () => void;
    text?: string;
}
