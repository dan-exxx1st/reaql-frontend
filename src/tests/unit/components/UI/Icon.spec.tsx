import React from 'react';
import { renderWithTheme } from 'tests/helpers/withTheme';

import { Icon } from 'components/UI';

describe('<Icon />', () => {
    describe('Snapshots', () => {
        it('Should render a person icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="person" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render an animal icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="animal" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render a food icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="food" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render a people icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="people" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render a search icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="search" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render a newChat icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="newChat" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render a threeDotsWhite icon  correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="threeDotsWhite" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render a threeDotsGrey icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="threeDotsGrey" />);

            expect(firstChild).toMatchSnapshot();
        });

        it('Should render a bluePlus icon correctly', () => {
            const {
                container: { firstChild },
            } = renderWithTheme(<Icon iconName="bluePlus" />);

            expect(firstChild).toMatchSnapshot();
        });
    });
});
