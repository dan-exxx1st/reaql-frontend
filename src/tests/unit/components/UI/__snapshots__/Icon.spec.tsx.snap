// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Snapshots Should render a bluePlus icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="BluePlus.svg"
/>
`;

exports[`<Icon /> Snapshots Should render a food icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="Food.svg"
/>
`;

exports[`<Icon /> Snapshots Should render a newChat icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="NewChat.svg"
/>
`;

exports[`<Icon /> Snapshots Should render a people icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="People.svg"
/>
`;

exports[`<Icon /> Snapshots Should render a person icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="Person.svg"
/>
`;

exports[`<Icon /> Snapshots Should render a search icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="SearchIcon.svg"
/>
`;

exports[`<Icon /> Snapshots Should render a threeDotsGrey icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="ThreeDotsGray.svg"
/>
`;

exports[`<Icon /> Snapshots Should render a threeDotsWhite icon  correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="ThreeDotsWhite.svg"
/>
`;

exports[`<Icon /> Snapshots Should render an animal icon correctly 1`] = `
.c0 {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0;
  margin: 0;
}

<img
  alt="Icon component"
  class="c0"
  src="Animal.svg"
/>
`;
